sqlglot/dialects/duckdb.py

```diff
@@ -77,8 +87,9 @@ def _build_sort_array_desc(args: t.List) -> exp.Expression:
     return exp.SortArray(this=seq_get(args, 0), asc=exp.false())
 
 
-def _build_date_diff(args: t.List) -> exp.Expression:
-    return exp.DateDiff(this=seq_get(args, 2), expression=seq_get(args, 1), unit=seq_get(args, 0))
+# Parse part-based functions, check https://duckdb.org/docs/sql/functions/date.html#date-functions
+def _build_date_part_exp(exp_class: t.Type[E], args: t.List) -> exp.Expression:
+    return (exp_class)(this=seq_get(args, 2), expression=seq_get(args, 1), unit=seq_get(args, 0))
```

Comment:
I'd rename this to `_build_date_delta` (the "delta" part is commonly used in sqlglot for these functions). Also, can we make this return a closure for consistent style? So then you'd eg. do this in `TRANSFORMS`:

```python
"DATEDIFF": _build_date_part_exp(exp.DateDiff)
```